neutron/diffraction/convert_with_calibration.cpp

```diff
@@ -33,6 +33,11 @@ template <class T> T convert_with_calibration_impl(T d, Dataset cal) {
     const auto &detector_info =
         d.labels()["detector_info"].template value<Dataset>();
     cal = merge(detector_info, cal);
+    // Masking and grouping information in the calibration table interferes with
+    // `groupby.mean`, dropping.
+    for (const auto &name : {"mask", "group"})
+      if (cal.contains(name))
+        cal.erase(name);
```

Comment:
Is if this serves no purpose would it be better to remove at the point of creating the calibration table?

Reply:
No, it essential and needed to masks pixels and define focussing groups in powder diffraction. I do however consider to eventually remove the grouping step from this algorithm and requiring that it is done externally. This would make more sense in terms of "one algorithm does one thing".